libvast/vast/address_synopsis.hpp

```diff
@@ -48,6 +54,66 @@ class address_synopsis final
   }
 };
 
+/// A synopsis for IP addresses that stores a copy of the input to
+/// be able to construct a smaller bloom filter from this data at
+/// some point using the `shrink` function.
+template <class HashFunction>
+class buffered_address_synopsis
+  : public bloom_filter_synopsis<address, HashFunction> {
+public:
+  using super = bloom_filter_synopsis<address, HashFunction>;
+
+  buffered_address_synopsis(type x, typename super::bloom_filter_type bf)
+    : super{std::move(x), std::move(bf)} {
+  }
+
+  void add(data_view x) override {
+    auto addr_view = caf::get_if<view<address>>(&x);
+    VAST_ASSERT(addr_view);
+    ips_.push_back(*addr_view);
+    super::add(x);
```

Comment:
Add a comment explaining the tradeoffs between using the extra bloom filter for the lookup vs using a hash set and implementing lookup over that.